### PR TITLE
Get integration tests passing for Kubernetes

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Disable some integration tests on kubernetes/minikube
- Fix bug in state machine

## Why are we making these changes?
The remainder of the non-cli tests pass on Kubernetes, so we can get a build pipeline.

### CODE REVIEW

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -211,11 +212,19 @@ def test_mea_culpa_retries(self):
 
     def test_compute_cluster(self):
         settings_dict = util.settings(self.cook_url)
-        # For now, we only have one compute cluster. This could be wrong with future
-        # refactors, but for now, this is the only compute cluster we have, so we should
-        # expect it.
         expected_compute_cluster = settings_dict['compute-clusters'][0]['config']['compute-cluster-name']
-        expected_mesos_framework = settings_dict['compute-clusters'][0]['config']['framework-id']
+        compute_cluster_factory_fn = settings_dict['compute-clusters'][0]['factory-fn']
```
Comment (scrosby) [inline integration/tests/cook/test_basic.py]: This smells like something we should refactor out. 
def cook_uses_kubernetes(....)
def cook_users_mesos(...)

File: integration/tests/cook/test_basic.py [hunk 2]
```diff
@@ -211,11 +212,19 @@ def test_mea_culpa_retries(self):
...
+        if compute_cluster_factory_fn == 'cook.kubernetes.compute-cluster/factory-fn':
+            expected_compute_cluster_type = 'kubernetes'
+        elif compute_cluster_factory_fn == 'cook.mesos.mesos-compute-cluster/factory-fn':
+            expected_compute_cluster_type = 'mesos'
+        else:
+            expected_compute_cluster_type = 'UNKNOWN - TEST WILL FAIL'
+        expected_compute_type = settings_dict['compute-clusters'][0]['config']['compute-cluster-name']
+        if 'framework-id' in settings_dict['compute-clusters'][0]['config']:
```
Comment (scrosby) [inline integration/tests/cook/test_basic.py]: Why not use 
     settings_dict['compute-clusters'][0]['config'].get('framework-id',None)

File: integration/tests/cook/test_basic.py [hunk 3]
```diff
@@ -211,11 +212,19 @@ def test_mea_culpa_retries(self):
...
         expected_compute_cluster = settings_dict['compute-clusters'][0]['config']['compute-cluster-name']
-        expected_mesos_framework = settings_dict['compute-clusters'][0]['config']['framework-id']
+        compute_cluster_factory_fn = settings_dict['compute-clusters'][0]['factory-fn']
+        if compute_cluster_factory_fn == 'cook.kubernetes.compute-cluster/factory-fn':
+            expected_compute_cluster_type = 'kubernetes'
+        elif compute_cluster_factory_fn == 'cook.mesos.mesos-compute-cluster/factory-fn':
+            expected_compute_cluster_type = 'mesos'
+        else:
+            expected_compute_cluster_type = 'UNKNOWN - TEST WILL FAIL'
+        expected_compute_type = settings_dict['compute-clusters'][0]['config']['compute-cluster-name']
```
Comment (scrosby) [inline integration/tests/cook/test_basic.py]: This var seems both dead and with the wrong name.

File: integration/tests/cook/test_basic.py [hunk 4]
```diff
@@ -1116,6 +1126,7 @@ def instance_query():
         finally:
             util.kill_jobs(self.cook_url, job_specs)
 
+    @unittest.skipIf(util.using_minikube(), 'Minikube only has 1 node, and we don\'t (yet) have an alternative test environment')
```
[the remark this answers is not in the document]
Comment (pschorf) [reply integration/tests/cook/test_basic.py]: Renamed

File: integration/tests/cook/test_basic.py (hunk 1 above)
Comment (pschorf) [reply integration/tests/cook/test_basic.py]: Done

File: integration/tests/cook/test_basic.py [hunk 5]
```diff
@@ -211,11 +212,19 @@ def test_mea_culpa_retries(self):
...
-        # expect it.
         expected_compute_cluster = settings_dict['compute-clusters'][0]['config']['compute-cluster-name']
-        expected_mesos_framework = settings_dict['compute-clusters'][0]['config']['framework-id']
+        compute_cluster_factory_fn = settings_dict['compute-clusters'][0]['factory-fn']
+        if compute_cluster_factory_fn == 'cook.kubernetes.compute-cluster/factory-fn':
+            expected_compute_cluster_type = 'kubernetes'
+        elif compute_cluster_factory_fn == 'cook.mesos.mesos-compute-cluster/factory-fn':
+            expected_compute_cluster_type = 'mesos'
+        else:
+            expected_compute_cluster_type = 'UNKNOWN - TEST WILL FAIL'
```
[the remark this answers is not in the document]
Comment (pschorf) [reply integration/tests/cook/test_basic.py]: Done

File: integration/tests/cook/test_basic.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (pschorf) [reply integration/tests/cook/test_basic.py]: Done

Comment (pschorf) [reply integration/tests/cook/test_basic.py]: Done

File: integration/tests/cook/test_basic.py (hunk 2 above)
Comment (pschorf) [reply integration/tests/cook/test_basic.py]: Done

File: integration/tests/cook/test_multi_user.py [hunk 6]
```diff
@@ -63,7 +63,7 @@ def test_group_delete_permission(self):
                 util.kill_jobs(self.cook_url, [job_uuid], assert_response=False)
 
     def test_multi_user_usage(self):
-        users = self.user_factory.new_users(6)
+        users = self.user_factory.new_users(4)
```
[the remark this answers is not in the document]
Comment (pschorf) [reply integration/tests/cook/test_multi_user.py]: There wasn't enough room on the single minikube node to schedule all of these jobs. It seemed pretty harmless to lower it, so we can keep running the test.

**Pull request merged**